### DISCUSSION TITLE
How to constantly request data via API using a script? And how to get completed orders?

### DISCUSSION THREAD

**Discussion by geolectric** [etsy/open-api#756] [Q&A] [answered]
Let's say I wanted to keep a local database of all my shop transactions. How would I do this with the current API? It doesn't seem like I can because I need to manually give approval to the app when the tokens eventually expire?

Other APIs, like Shopify, seem to have an app password that you can use in place of a token, which I believe solves this issue.

For example, let's say I want to try dropship printing, I would need a script that watches for completed transactions, and then I would use a dropshipping API to place the order. Is there a better way to do this?

And one more related question, how do I get orders where the payment has been completed and was successful? Can I use getShopReceiptTransactionsByShop?

https://developers.etsy.com/documentation/reference/#operation/getShopReceiptTransactionsByShop

If it's in that response, does it mean the payment was completed and successful?

**Comment by etsyachristensen** [comment, marked as answer]
1. Every time you generate an OAuth 2 token, which is only valid for 1 hour, you also get a refresh_token that is valid for 90 days. You can use the refresh token to fetch a new auth token without having to go through the entire consent process. See the documentation here: https://developers.etsy.com/documentation/essentials/authentication#step-3-request-an-access-token

2. If you were to use a drop ship printing service, you may find that they already have integrations with Etsy and all you'd need to do is add their app to your shop. You'll have to check with the vendor to see if they have one. If not, you'd have to build an app that will consume the API and download your information to your local database and then push orders to the drop shipper.

3. Yes. In the Receipt and Transaction objects, you will find that they indicate the the order has been completed and successfully charged. At this point, most any shop will be using the Etsy integrated payments system so receipts aren't created until the order has been completely processed.

**Reply by geolectric** [reply]
Thanks @etsyachristensen, so just to clarify #1 ... even with the refresh token, at some point after the 90 days, then a manual approval (entering username/password) process would be needed again? Is that correct?

Could you briefly explain the difference between Shop Receipt and Shop Receipt Transactions? How are the two related? Or what would you use each one for? I'm having trouble wrapping my head around it. I think part of my issue is that my Shop is brand new and I don't have any sales records to view using the API.

**Reply by etsyachristensen** [reply]
No actually. Once a shop grants access to the app, unless they later revoke access via shop manager, you can simply request a new access token by passing the same grant information you originally used.

A shop receipt can have many transactions. If a user purchases multiple items for instance, there might be several transactions in a receipt.
